# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/TextDirectiveCreator.h
# Commit: 09616085d8ef
# Full Hash: 09616085d8ef6470309e1b3c565242d6a811cd5d
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-07-22 09:32:06
# Regressor Bug: 1978235
# File Overlap Count: 1
# Description:
#   Bug 1978235 - Text Fragments: Ensure that start and end terms are always non-empty. r=mccr8,dom-core
#   
#   The patch for Bug 1976473 changed  the behavior of the algorithm to collect word boundary distances
#   to merge punctuation characters with the preceding word. However, this led to unexpected behavior if a
#   context term only consisted of punctuation, where this function would unexpectedly return an empty array.
# ==============================================================================

diff -r df6c5c98b663 -r 09616085d8ef dom/base/TextDirectiveCreator.h
--- a/dom/base/TextDirectiveCreator.h	Mon Jul 21 18:33:07 2025 +0000
+++ b/dom/base/TextDirectiveCreator.h	Mon Jul 21 18:36:11 2025 +0000
@@ -88,8 +88,12 @@
    * @brief Collects text content surrounding the target range.
    *
    * The context terms are then stored both in normal and fold case form.
+   *
+   * Returns false if the algorithm cannot continue, for example if the text
+   * directive must use range-based matching because of its length, but the
+   * target range only consists of one word.
    */
-  virtual Result<Ok, ErrorResult> CollectContextTerms() = 0;
+  virtual Result<bool, ErrorResult> CollectContextTerms() = 0;
 
   /**
    * @brief Common helper which collects the prefix term of the target range.
@@ -114,14 +118,8 @@
    *
    * The distances are always sorted, so that the first entry points to the
    * nearest word boundary in search direction.
-   *
-   * This method returns false if collecting context term word boundary
-   * distances failed in a way that it's not considered a failure, but rather
-   * it's not possible to create a text directive for the target range.
-   * This can happen if the target range is too long for exact matching, but
-   * does not contain a word boundary.
    */
-  virtual bool CollectContextTermWordBoundaryDistances() = 0;
+  virtual void CollectContextTermWordBoundaryDistances() = 0;
 
   /**
    * @brief Searches the document for other occurrences of the target range and
@@ -254,9 +252,9 @@
  private:
   using TextDirectiveCreator::TextDirectiveCreator;
 
-  Result<Ok, ErrorResult> CollectContextTerms() override;
+  Result<bool, ErrorResult> CollectContextTerms() override;
 
-  bool CollectContextTermWordBoundaryDistances() override;
+  void CollectContextTermWordBoundaryDistances() override;
 
   Result<Ok, ErrorResult> FindAllMatchingCandidates() override;
 
@@ -304,9 +302,9 @@
  private:
   using TextDirectiveCreator::TextDirectiveCreator;
 
-  Result<Ok, ErrorResult> CollectContextTerms() override;
+  Result<bool, ErrorResult> CollectContextTerms() override;
 
-  bool CollectContextTermWordBoundaryDistances() override;
+  void CollectContextTermWordBoundaryDistances() override;
 
   Result<Ok, ErrorResult> FindAllMatchingCandidates() override;
 